src/generators/modelsGenerator.ts

```diff
@@ -444,13 +444,17 @@ function writeUniontypes({ objects }: ClientDetails, modelsFile: SourceFile) {
     .forEach(obj => {
       const polymorphicObject = obj as PolymorphicObjectDetails;
       const childrenNames = [
-        obj.name,
         ...polymorphicObject.children.map(c => {
           return c.schema.children && c.schema.children.immediate.length
             ? `${c.name}Union`
             : c.name;
         })
       ];
+
+      if (isPolymorphicParentInUnion(polymorphicObject)) {
+        childrenNames.unshift(polymorphicObject.name);
+      }

```

Comment:
Nit: I am not sure about the complexity of `unshift`. Perhaps we can do this check first to initialize the array and then add  the children names after.

Reply:
Yeah, I'll do that